### PR TITLE
(Readability) Replace all instances of suppress_ast with emit_ast to avoid negated logic

### PR DESCRIPTION
**Pull request by sfc-gh-oplaton**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-0000000

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

Stylistic change - prefer direct logic instead of inverted logic.

Use `emit_ast` (default = `True`) everywhere instead of `suppress_ast` (default = `False`).

To get ASTs out of public APIs, do nothing. To prevent public APIs from emitting ASTs, provide `emit_ast = False`.

Name the argument `_emit_ast` on public APIs and `emit_ast` on internal APIs.

### CODE REVIEW

File: src/snowflake/snowpark/dataframe.py [hunk 1]
```diff
@@ -277,15 +277,15 @@ def _disambiguate(
             )
             for name in lhs_names
         ],
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/dataframe.py:279]: this change makes sense to me, let's also check that all docstrings are updated with a unique description (that probably needs to get copied around). If you dislike copy&paste of the docstring, we could think of a decorator to inject a unique description. For mid-term, we may need to think about using a PIMPL pattern for the actual implementation. Most of the current Snowpark APIs are public, so adding a new parameter could be problematic.

**Review by sfc-gh-lspiegelberg — APPROVED** [review]
LGTM, let's check docstrings as well before merging.

**Review by sfc-gh-azwiegincew — APPROVED** [review]

File: src/snowflake/snowpark/session.py [hunk 2]
```diff
@@ -1834,17 +1834,15 @@ def update_query_tag(self, tag: dict) -> None:
                     f"Expected query tag to be valid json. Current query tag: {tag_str}"
                 )
 
-    def table(
-        self, name: Union[str, Iterable[str]], suppress_ast: bool = False
-    ) -> Table:
+    def table(self, name: Union[str, Iterable[str]], emit_ast: bool = True) -> Table:
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/session.py:1837]: nit: table should be public API? so `_emit_ast` according to PR description?

File: src/snowflake/snowpark/table.py [hunk 3]
```diff
@@ -273,9 +273,9 @@ def __init__(
         table_name: str,
         session: Optional["snowflake.snowpark.session.Session"] = None,
         ast_stmt: Optional[proto.Assign] = None,
-        suppress_ast: bool = False,
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/table.py:276]: nit: same here, should be public. tbh, I am ok to call it always `_emit_ast`. Will make things easier.

Comment (sfc-gh-oplaton) [reply src/snowflake/snowpark/table.py:276]: Changed all files to use `_emit_ast` uniformly.

File: src/snowflake/snowpark/dataframe.py (hunk 1 above)
Comment (sfc-gh-oplaton) [reply src/snowflake/snowpark/dataframe.py:279]: Updated all docstrings to use the same description.

File: src/snowflake/snowpark/session.py (hunk 2 above)
Comment (sfc-gh-oplaton) [reply src/snowflake/snowpark/session.py]: Done.

**Pull request merged**